diff_drive_controller/test/diff_drive_nan_test.cpp

```diff
@@ -93,25 +93,25 @@ TEST_F(DiffDriveControllerTest, testNaNCmd)
   ros::Duration(0.1).sleep();
 
   // send NaN
-  for(int i = 0; i < 10; ++i)
+  for (int i = 0; i < 10; ++i)
   {
     geometry_msgs::Twist cmd_vel;
     cmd_vel.linear.x = NAN;
     cmd_vel.angular.z = NAN;
     publish(cmd_vel);
     geometry_msgs::TwistStamped odom_msg = getLastCmdVelOut();
-    EXPECT_EQ(std::isnan(odom_msg.twist.linear.x), false);
-    EXPECT_EQ(std::isnan(odom_msg.twist.angular.z), false);
+    EXPECT_FALSE(std::isnan(odom_msg.twist.linear.x));
+    EXPECT_FALSE(std::isnan(odom_msg.twist.angular.z));
     ros::Duration(0.1).sleep();
   }
 
   nav_msgs::Odometry odom = getLastOdom();
-  EXPECT_EQ(odom.twist.twist.linear.x, 0.0);
-  EXPECT_EQ(odom.pose.pose.position.x, 0.0);
-  EXPECT_EQ(odom.pose.pose.position.y, 0.0);
+  EXPECT_DOUBLE_EQ(odom.twist.twist.linear.x, 0.0);
+  EXPECT_DOUBLE_EQ(odom.pose.pose.position.x, 0.0);
+  EXPECT_DOUBLE_EQ(odom.pose.pose.position.y, 0.0);
```

Comment:
I would always use EXPECT_NEAR to compare double values.

Reply:
This is pretty much that with a tiny tiny epsilon
https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#floating-point-comparison